telemetry/internal/appinsights/throttle.go

```diff
@@ -0,0 +1,142 @@
+package appinsights
+
+import (
+	"time"
+)
+
+type throttleManager struct {
+	msgs chan *throttleMessage
+}
+
+type throttleMessage struct {
+	query     bool
+	wait      bool
+	throttle  bool
+	stop      bool
+	timestamp time.Time
+	result    chan bool
+}
+
+func newThrottleManager() *throttleManager {
```

Comment:
Would be nice to have some documentation about the intent of this struct and its methods.

Reply:
I've removed this struct altogether. Too complicated with little benefit.